### DISCUSSION TITLE
How to keep images aligned within the max width?

### DISCUSSION THREAD

**Discussion by imnithish** [varabyte/kobweb#449] [Q&A] [answered]
So, I have a box having a max width:

`Box(
        modifier = Modifier
            .fillMaxWidth()
            .background(Theme.Bg.rgb)
            .maxWidth(Constants.SECTION_WIDTH.px),
    ) {
Column{Content()}
    }`

The  content composable contains a row with an image and column containing some texts.

`@Composable
fun Content() {
    Row(
        modifier = Modifier.fillMaxWidth().margin(top = 109.px, left = 89.px),
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        Column {
            /**
             * ...
             */
        }
        Image(
            src = Res.Image.image_4
        )
    }  
}`

The issue is that the image is going out of the Box scope. It can be seen if scrolled horizontally. How can i keep the image within the Box itself?
 
 
<img width="1412" alt="Screenshot 2024-01-28 at 11 32 00 PM" src="https://github.com/varabyte/kobweb/assets/19818737/1ea09551-4570-4485-9377-c1a16d66e1a4">

Thank you!

**Comment by bitspittle** [comment]
I'll be honest -- layout stuff like this is surprisingly hard and frustrating with web. The way width calculations are frustrating for sure. I am pretty sure the sizing fight you're witnessing here is ultimately one of dealing with legacy decisions of the early web, although I'm not 100% sure :)

If you want a 50/50 split here, you might try a `SimpleGrid` with num columns set to two. Otherwise, you might want to explicitly set the size of the Image, either to a percent of the row or an absolute value, just to see if it at least fixes the issue.

**Reply by imnithish** [reply]
Unfortunately, both the suggestion you gave is not working :(.
This is really frustrating. It feels like a simple thing to do, but I can't find a solution for this.

**Reply by bitspittle** [reply]
I don't disagree that it's frustrating. 

What I recommend is open up dev tools for your page, find your element that isn't sizing right, and then make sure the width for the element and every single one of its parents are set. See if setting widths anywhere fixes the problem.

Basically the way web works is if you specify a width, it uses it, but if you don't, it tries to figure it out automatically, but that automatic logic is complex and sometimes counterintuitive. Sometimes the fix is making sure every parent all the way from body to child has width set to 100% (or Modifier.fillMaxWidth)

**Reply by imnithish** [reply, marked as answer]
I replaced Row( modifier = Modifier.fillMaxWidth().margin(top = 109.px, left = 89.px) with Row( modifier = Modifier.fillMaxWidth().padding(top = 109.px, left = 89.px). Now, it's working fine. 😁

Thank you!
